processing/src/main/java/org/apache/druid/query/filter/FilterTuning.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class FilterTuning
+{
+  public static FilterTuning createDefault(boolean useIndex)
+  {
+    return new FilterTuning(useIndex, 0, Integer.MAX_VALUE);
+  }
+
+  private final Boolean useIndex;
```

Comment:
Please mark these `@Nullable` too, just like the constructor parameters.

Or, apply the defaults you have in `getUseIndex()`, etc, in the constructors and make these fields primitives.

Reply:
applied defaults in constructor